### PR TITLE
BN-292 and BN-293

### PR DESCRIPTION
**Pull request by Jaga-A**

### CODE REVIEW

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class KafkaProducerService {
+    private final ObjectMapper objectMapper;
+
+    private static final String DOMAIN_EVENT_TYPE = "SIMPLE_METRIC";
+    private static final String DOMAIN_EVENT_SOURCE = "daily-indexing";
+    private static final String DOMAIN_EVENT_ACTOR = "indexer";
+    private static final String ORG_ID = "1";
```
Comment (AiswaryaPappusamy97) [inline src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:38]: Is it ok to set 1?

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                .collect(Collectors.toList());
+
+    }
+
+    private MpMetricDomainEvent getIgMediaPayload(IgIndexingSucceeded payload) {
+        try {
+            final IgMediaPayload media = objectMapper.treeToValue(payload.getDataPayload(), IgMediaPayload.class);
+            Map<String, String> metaData = new HashMap<>();
+            metaData.put("post_id", media.getId());
+            metaData.put("type", media.getMediaType());
```
Comment (AiswaryaPappusamy97) [inline src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:61]: this mediaType will return `IMAGE` / `VIDEO` . Should we set that here?

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+            return MpMetricDomainEvent.builder().id(UUID.randomUUID()).type(DOMAIN_EVENT_TYPE).actorId(DOMAIN_EVENT_ACTOR)
+                    .payload(MpSimpleMetricEvent.builder()
+                            .meta(MpMetricEventMeta.builder()
+                                    .context(ORG_ID)
+                                    .bucket(stripHyphensFromChannelUuid(payload.getChannelUUID().toString()))
+                                    .additionalMeta(metaData)
+                                    .build())
+                            .timestamp(LocalDateTime.ofInstant(media.getTimestamp(), ZoneOffset.UTC))
```
Comment (AiswaryaPappusamy97) [inline src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:70]: Please check if this the timestamp of the post / the timestamp on which the data is fetched from API

**Review by AiswaryaPappusamy97 — CHANGES_REQUESTED** [review]
Please add tests

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java (hunk 1 above)
Comment (Jaga-A) [reply src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:38]: Hardcoded it. Just for POC.

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java (hunk 3 above)
Comment (Jaga-A) [reply src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:70]: post created date they are using. After poc we can tackle this

**Review by AiswaryaPappusamy97 — APPROVED** [review]

File: src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java (hunk 2 above)
Comment (Jaga-A) [reply src/main/java/io/falcon/benchmark/measure/kafka/service/KafkaProducerService.java:61]: not sure. tackle it after poc

**Pull request merged**